### PR TITLE
fix: Incorrect results when `ParseQuery` contains special characters

### PR DESCRIPTION
**Pull request by Nidal-Bakir**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description

Closes: #841

### Approach
use `Uri.encodeComponent(URLComponent);`

This will produce an encoded query value:
`?where={"Name":"AB%2BAB"}`
The encoded equivalent of:
`?where={"Name": "AB+AB"}`

### TODOs before merging

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry

### CODE REVIEW

**Comment by Nidal-Bakir** [comment]
I will await my other PR (#860) to be merged to see all the added tests and changes.

**Comment by Nidal-Bakir** [comment]
@parse-community/flutter-sdk-review 

This is ready.

**Review by mbfakourii — APPROVED** [review]

**Review by mbfakourii — APPROVED** [review]
Everything looks good

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [5.1.2](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-5.1.1...dart-5.1.2) (2023-05-27)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [5.1.2](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-5.1.1...dart-5.1.2) (2023-05-29)
```

File: packages/dart/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
+
+### Bug Fixes
+
+* ParseQuery does not return the correct result when a query contains special characters ([#866](https://github.com/parse-community/Parse-SDK-Flutter/pull/866))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
* Incorrect results when `ParseQuery` contains special characters ([#866](https://github.com/parse-community/Parse-SDK-Flutter/pull/866))
```

**Review by mtrezza — APPROVED** [review]
Looks good!

**Pull request merged**